tests/trainer/test_dataloaders.py

```diff
@@ -746,6 +746,39 @@ def test_dataloader_distributed_sampler(tmpdir):
     trainer.test(ckpt_path=None)
 
 
+@pytest.mark.skipif(platform.system() == 'Windows', reason='Does not apply to Windows platform.')
+@pytest.mark.skipif(torch.cuda.device_count() < 2, reason='Test requires multiple GPUs')
+def test_dataloader_distributed_sampler_already_attached(tmpdir):
+    """ Test DistributedSampler and it's arguments for DDP backend when DistSampler already included on dataloader """
+
+    class ModelWithDataLoaderDistributedSampler(EvalModelTemplate):
```

Comment:
hey @ananthsub 
Looks like you need to move this class outside the test function or we get the good old pickling error :)